src/webapp/components/landing-page-edit-dialog/LandingPageEditDialog.tsx

```diff
@@ -111,11 +114,40 @@ export const LandingPageEditDialog: React.FC<LandingPageEditDialogProps> = props
     };
 
     const handleFileUpload = useCallback(
-        (event: ChangeEvent<HTMLInputElement>) => {
+        (event: ChangeEvent<HTMLInputElement>, fileType: keyof LandingNode) => {
             const file = event.target.files ? event.target.files[0] : undefined;
+
             file?.arrayBuffer().then(async data => {
+                const reader = new FileReader();
+                reader.onload = e => {
+                    const img = new Image();
+                    img.onload = () => {
+                        const width = img.width;
+                        const height = img.height;
+                        const aspectRatio = width / height;
+
+                        setWarnings([]);
+                        const newWarnings: string[] = [];
+
+                        if (fileType === "favicon") {
+                            if (aspectRatio !== FAVICON_ASPECT_RATIO) {
+                                newWarnings.push("Please ensure that your favicon has a 1:1 aspect ratio.");
+                            }
+
+                            if (width > FAVICON_MAX_SIZE || height > FAVICON_MAX_SIZE) {
+                                newWarnings.push("Please use an icon of 128x128 pixels or smaller.");
+                            }
+
+                            newWarnings.length !== 0 && setWarnings(newWarnings);
```

Comment:
Using functional programming to assign directly the warnings you will no need to make this check later